spec/javascripts/directives/scheduler/updateDropdownForFilter_spec.js

```diff
@@ -6,17 +6,17 @@ describe('update-drop-down-for-filter initialization', function() {
     var element = angular.element(
       '<form name="angularForm">' +
       '<select id="filter_typ" name="filter_typ" ng-model="scheduleModel.filter_typ"><option>All VMs</option><option>Ketchup</option><option>Relish</option></select>' +
-      '<select id="filter_value" name="filter_value" update-dropdown-for-filter dropdown-model="scheduleModel" dropdown-list="filterList" filter-hide="filterValuesEmpty" ng-model="scheduleModel.filter_value"><option value="" label="Choose">Choose</option><option value="0" label="Mustard">Mustard</option><option value="1" label="Ketchup">Ketchup</option><option value="2" label="Relish">Relish</option>' +
+      '<select id="filter_value" name="filter_value" update-dropdown-for-filter dropdown-model="scheduleModel" dropdown-list="filterList" filter-hide="filterValuesEmpty" ng-model="scheduleModel.filter_value"><option disabled="" value="" class="" selected="selected">Choose</option><option value="string:Mustard" label="Mustard">Mustard</option><option value="string:Ketchup" label="Ketchup">Ketchup</option><option value="string:Relish" label="Relish">Relish</option>' +
```

Comment:
>> @AparnaKarve, are you sure that's a thing?
@himdel You'd be surprised. Check this out - 
http://www.devixgroup.com/blog/2015/05/migrating-angularjs-1-3-to-1-4/
Check the section under ```ng-options```

Here's another article -
http://gurustop.net/blog/2015/06/23/angularjs-1_4-select-track_by-key

For the record, ```ng-options``` is already being used.

Per the above article , when ```ng-options``` is used, Angular 1.4 will change the html from -
```
<select ng-model="x" ng-option="i in items">
  <option value="1">a</option>
  <option value="2">b</option>
  <option value="3">c</option>
  <option value="4">d</option>
</select>
```

to

```
<select ng-model="x" ng-option="i in items">
  <option value="string:a">a</option>
  <option value="string:b">b</option>
  <option value="string:c">c</option>
  <option value="string:d">d</option>
</select>
```
And then there is something new introduced in ```Angular 1.4``` called ```track by``` to be used with ```select ng-options``` with which we can avoid dealing with an ```option value``` that looks like this ```string:abc```.

Bottomline, the change in the spec was merely done to respond to the changes in ```Angular 1.4.3```.

That being said, I am now going to give ```track by``` a try and find out if we can avoid dealing with the weird construct in ```option value```